test/unit/python/test_write_init_files.py

```diff
@@ -258,7 +258,90 @@ def test_simple_reg_constituent_write_init(self):
+
+    def test_simple_initial_value_write_init(self):
+        """
+        Test that the 'write_init_files' function
+        generates the correct Fortran code given
+        a simple registry and CCPP physics suite with
+        only regular variables with one having an initial_value set.
+        """
+
+        # Setup registry inputs:
+        filename = os.path.join(_INIT_SAMPLES_DIR, "simple_reg_initial_value.xml")
+        out_source_name = "physics_types_simple"
+        out_source = os.path.join(_TMP_DIR, out_source_name + '.F90')
+        out_meta = os.path.join(_TMP_DIR, out_source_name + '.meta')
+
+        # Setup capgen inputs:
+        model_host = os.path.join(_INIT_SAMPLES_DIR,"simple_host.meta")
+        sdf = os.path.join(_INIT_SAMPLES_DIR,"suite_simple.xml")
+        scheme_files = os.path.join(_INIT_SAMPLES_DIR, "temp_adjust.meta")
+        cap_datafile = os.path.join(_TMP_DIR, "datatable_simple.xml")
+
+        host_files = [model_host, out_meta]
+
+        # Setup write_init_files inputs:
+        vic_name = "phys_vars_init_check_initial_value.F90"
+        pi_name = "physics_inputs_initial_value.F90"
+        check_init_out = os.path.join(_TMP_DIR, vic_name)
+        phys_input_out = os.path.join(_TMP_DIR, pi_name)
+        # Setup comparison files
+        check_init_in = os.path.join(_INIT_SAMPLES_DIR, vic_name)
+        phys_input_in = os.path.join(_INIT_SAMPLES_DIR, pi_name)
+
+        # Create local logger:
+        logger = logging.getLogger("write_init_files_initial_value")
+
+        # Clear all temporary output files:
+        remove_files([out_source, out_meta, cap_datafile,
+                      check_init_out, phys_input_out])
+
+        # Generate registry files:
+        _, _, ic_names, _, vars_init_value = gen_registry(filename, 'se', _TMP_DIR, 3,
+                                                          _SRC_MOD_DIR, _CAM_ROOT,
+                                                          loglevel=logging.ERROR,
+                                                          error_on_no_validate=True)
+
+        # Generate CCPP capgen files:
+        kind_types = ['kind_phys=REAL64']
+        run_env = CCPPFrameworkEnv(logger, host_files=host_files,
+                                   scheme_files=scheme_files, suites=sdf,
+                                   preproc_directives='',
+                                   generate_docfiles=False,
+                                   host_name='cam', kind_types=kind_types,
+                                   use_error_obj=False,
+                                   force_overwrite=True,
+                                   output_root=_TMP_DIR,
+                                   ccpp_datafile=cap_datafile)
+
+        cap_database = capgen(run_env, return_db=True)
+
+        # Generate physics initialization files:
+        retmsg = write_init.write_init_files(cap_database, ic_names, [], ["air_pressure_at_sea_level"], _TMP_DIR,
```

Comment:
Instead of hard-coding `air_pressure_at_sea_level` here I would use the `vars_init_value` passed out by `gen_registry`:

```suggestion
        retmsg = write_init.write_init_files(cap_database, ic_names, [], vars_init_value, _TMP_DIR,
```